lua/lspconfig/volar.lua

```diff
@@ -0,0 +1,77 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local function get_typescript_server_path(root_dir)
+  local project_root = util.find_node_modules_ancestor(root_dir);
+  return project_root and (util.path.join(project_root, 'node_modules', 'typescript', 'lib', 'tsserverlibrary.js')) or '';
+end
+
+local server_name = 'volar'
+local bin_name = 'volar-server'
+
+-- https://github.com/johnsoncodehk/volar/blob/master/packages/shared/src/types.ts
+local volar_init_options = {
+  typescript = {
+    serverPath = get_typescript_server_path(vim.fn.getcwd()),
```

Comment:
this vim.fn.getcwd seems brittle, should it not use rootdir? This can be done in on_new_config if necessary.

Reply:
This comes from https://github.com/neovim/nvim-lspconfig/blob/08a698d652f3d1fc6195b4efd9d23ecaced2c879/lua/lspconfig/angularls.lua#L15 , but perhaps it's warranted in case of angularls's probedir but weird for detecting typescript.

I'll try with on_new_config later

Reply:
Done https://github.com/neovim/nvim-lspconfig/pull/1223/commits/f170b0cbf23ff10efb4ee0755259befd29fff263

Reply:
I'm not sure if straight up not providing serverPath in init_options to Volar is a mistake. Anyhow, I preferred the user to get an error from Volar in that case rather than get some vague lua error about "index is nil".

Hence the if chain check